tests/functional_tests/0006_set_test.py

```diff
@@ -22,21 +22,73 @@
 # IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 import pytest
-from click.testing import CliRunner
 
-import iocage_cli as ioc
 
 require_root = pytest.mark.require_root
 require_zpool = pytest.mark.require_zpool
 
 
+# Let's perform testing first on different types of jails
+# Then we can move on to setting some specific properties and seeing
+# do the props work as intended
+
+
+# TODO: Plugin test left
+
+
+def _set_and_test_note_prop(invoke_cli, value, config_data, jail):
+    invoke_cli(
+        ['set', f'notes={value}', jail]
+    )
+
+    assert (config_data(jail) or {}).get('notes') == value,\
+        f'Failed to set note value to {value}'
+
+
+@require_root
+@require_zpool
+def test_01_set_prop_on_thin_jail(thin_jails, invoke_cli, config_data):
+    if thin_jails:
+        _set_and_test_note_prop(
+            invoke_cli, 'thin_jail', config_data, list(thin_jails.keys())[0]
+        )
+
+
 @require_root
 @require_zpool
-def test_set():
-    jails = ["771ec0cf-afdd-455d-9245-4a890e228325", "dfb013e5"]
-    runner = CliRunner()
+def test_02_set_prop_on_thick_jail(thick_jails, invoke_cli, config_data):
+    if thick_jails:
+        _set_and_test_note_prop(
+            invoke_cli, 'thick_jail', config_data, list(thick_jails.keys())[0]
+        )
 
-    for jail in jails:
-        result = runner.invoke(ioc.cli, ["set", f"notes={jail}", jail])
 
-        assert result.exit_code == 0
+@require_root
+@require_zpool
+def test_03_set_prop_on_thick_config_jail(
+        thick_config_jails, invoke_cli, config_data
+):
+    if thick_config_jails:
+        _set_and_test_note_prop(
+            invoke_cli, 'thick_config_jail', config_data,
```

Comment:
Let's test a prop with spaces and quotes for notes, that has historically been an issue.


```suggestion
            invoke_cli, 'foo \"bar\"', config_data,
```